keyStore.go

```diff
@@ -26,7 +27,13 @@ func SetUp(key string) {
 	keyLock.Lock()
 	durationLock.Lock()
 	delete(keyState, key)
-	delete(keyDurations, key)
+
+	// support GetAndClearLastUpTime
+	lastDown := keyDurations[key]
```

Comment:
I like the concept but I wonder if this is a feature flag type situation or not.

The only edge case I can think of is multiple pseudo keypresses that somehow come in with 0 time delta.

I dont think this will break anyone's implementation but if it is a real break we should consider if this becomes release/3.0.0 feature.

Reply:
It's not a breaking change. If the lastDown is zero it doesn't need to be deleted. So no change to users. Only added functionality.

Treating multiple very fast keypresses of the same key as one is indeed part of the feature that is added.